src/System.IO.FileSystem/tests/Directory/Move.cs

```diff
@@ -212,6 +212,57 @@ public void Path_Longer_Than_MaxLongPath_Throws_Exception()
             });
         }
 
+        [Fact]
+        public void ThrowIOExceptionWhenDirectoryExists()
+        {
+            Directory.CreateDirectory($"{TestDirectory}/foo");
+            Assert.Throws<IOException>(() => Directory.Move($"{TestDirectory}/foo", $"{TestDirectory}/foo"));
+        }
+
+        [Fact]
+        public void MoveDirectoryToNewDirectoryButWithDifferentCasing()
+        {
+            Directory.CreateDirectory(Path.Combine(TestDirectory, "foo"));
```

Comment:
I would add tests that don't match the actual file system casing. What happens when you have `FOO` and you try to move `foo` to `FOO`?

Reply:
Is this addressed?

Reply:
@JeremyKuhne just so I understand, you mean I have a directory of `FOO` and I ask to move a directory `foo` to `FOO`. So sending a case-variant as an argument into Move?

Reply:
@Jlalond yes "FOO" exists as a directory and you try to move "foo" (which also exists) to "FOO". (Sorry about the slow response.) 🙏

Reply:
I believe I addressed this starting [299](https://github.com/dotnet/corefx/pull/40570/files#diff-4b86f8ca99b5dc228df21c6e6e417034R299) can you confirm my logic? Specifically around the case-sensitive vs insensitive OS

Reply:
I think we're ok here.